scripts/scil_uniformize_streamlines_endpoints.py

```diff
@@ -0,0 +1,97 @@
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             add_verbose_arg,
+                             assert_outputs_exist,
+                             assert_inputs_exist)
+from scilpy.utils.streamlines import get_centroid_streamline
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+    p.add_argument('in_bundle',
+                   help='Input path of the tractography file.')
+    p.add_argument('out_bundle',
+                   help='Output path of the uniformized file.')
+
+    method = p.add_mutually_exclusive_group(required=True)
+    method.add_argument('--axis', choices=['x', 'y', 'z'],
+                        help='Match endpoints of the streamlines along this axis.\n'
+                        'SUGGESTION: Commissural = x, Association = y, '
+                        'Projection = z')
+    method.add_argument('--auto', action='store_true',
+                        help='Match endpoints of the streamlines along an '
+                             'automatically determined axis.')
+
+    p.add_argument('--swap', action='store_true',
+                   help='Swap head <-> tail convention. '
+                        'Can be useful when the reference is not in RAS.')
+    add_reference_arg(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.INFO)
+
+    assert_inputs_exist(parser, args.in_bundle)
+    assert_outputs_exist(parser, args, args.out_bundle)
+
+    sft = load_tractogram_with_reference(parser, args, args.in_bundle)
+    axis = ['x', 'y', 'z']
+    if args.auto:
+        centroid = get_centroid_streamline(sft.streamlines, 20)[0]
+        main_dir_ends = np.argmax(np.abs(centroid[0] - centroid[-1]))
+        main_dir_displacement = np.argmax(np.abs(np.sum(np.gradient(centroid,
+                                                                    axis=0),
+                                                        axis=0)))
+        if main_dir_displacement != main_dir_ends:
+            logging.info('Ambiguity in orientation, you should use --axis')
+        args.axis = axis[main_dir_displacement]
+        logging.info('Orienting endpoints of {} in the {} axis'.format(
+            args.in_bundle, args.axis))
+
+    axis_pos = axis.index(args.axis)
+    for i in range(len(sft.streamlines)):
+        if sft.streamlines[i][0][axis_pos] > sft.streamlines[i][-1][axis_pos]:
+            sft.streamlines[i] = sft.streamlines[i][::-1]
+            for key in sft.data_per_point[i]:
+                sft.data_per_point[key][i] = sft.data_per_point[key][i][::-1]
+        if args.swap:
+            sft.streamlines[i] = sft.streamlines[i][::-1]
```

Comment:
if args.swap, you flip on line 85 and then re-flip here. Not super efficient. Maybe there's a better way? Something like:

```
if args.swap and wrong_axis:
    ...
elif not args.swap and wrong_inversed_axis:
    ...
```